### PR TITLE
Add documentation for existing resources

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
This adds documentation for the following resources:

  - aws_ec2_instance
  - aws_iam_access_key
  - aws_iam_password_policy
  - aws_iam_root_user
  - aws_iam_users

Signed-off-by: Jerry Aldrich <jerryaldrichiii@gmail.com>

### CODE REVIEW

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
Need to document resource-specific matchers (like aws_ec2_instance - be_running)

**Comment by jerryaldrichiii** [comment]
That should do it! Great suggestion. I also modified the doc you created to match mine.

Let me know if you want anything else :smile:

File: docs/resources/aws_ec2_instance.md [hunk 1]
```diff
@@ -54,4 +54,46 @@ The following examples show how to use this InSpec audit resource.
 
 ## Matchers
 
-For a full list of available matchers (such as `be_running`) please visit our [matchers page](https://www.inspec.io/docs/reference/matchers/).
+This InSpec audit resource has the following special matchers. For a full list of available matchers (such as `exist`) please visit our [matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+### be_pending
+
+The `pending` matcher tests if the described EC2 instance state is `pending`
```
Comment (clintoncwolfe) [inline docs/resources/aws_ec2_instance.md]: Be sure to include be_ on predicates in text

File: docs/resources/aws_ec2_instance.md [hunk 2]
```diff
@@ -54,4 +54,46 @@ The following examples show how to use this InSpec audit resource.
...
+
+    it { should be_pending }
+
+### be_running
+
+The `be_running` matcher tests if the described EC2 instance state is `running`
+
+    it { should be_running }
+
+### be_shutting_down_
```
Comment (clintoncwolfe) [inline docs/resources/aws_ec2_instance.md]: Trailing '_' typo

File: docs/resources/aws_ec2_instance.md [hunk 3]
```diff
@@ -54,4 +54,46 @@ The following examples show how to use this InSpec audit resource.
...
+
+The `be_shutting_down` matcher tests if the described EC2 instance state is `shutting-down`
```
Comment (clintoncwolfe) [inline docs/resources/aws_ec2_instance.md]: I'm wondering - some states are valid, but very transitory.  shutting_down, pending, stopping - those might just be mentioned with less prose, as they are much less likely to be of use.

**Comment by jerryaldrichiii** [comment]
Great catch @clintoncwolfe! Should be good now!

File: docs/resources/aws_iam_users.md [hunk 4]
```diff
@@ -38,8 +38,26 @@ The following examples show how to use this InSpec audit resource.
       it { should exist }
     end
 
+### Test that all users that have a console password have Multi-Factor Authentication enabled
+
+    describe aws_iam_users.where(has_console_password?: true) do
```
Comment (clintoncwolfe) [inline docs/resources/aws_iam_users.md]: Just a note, I made #103 to document `where` and the various things you can pass to it.  We'll address in a future PR.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]

File: docs/resources/aws_ec2_instance.md (hunk 2 above)
Comment (jerryaldrichiii) [reply docs/resources/aws_ec2_instance.md]: Fixed! :sweat_smile:

File: docs/resources/aws_ec2_instance.md (hunk 3 above)
Comment (jerryaldrichiii) [reply docs/resources/aws_ec2_instance.md]: Indeed, I was wondering about that when I was documenting them. Only thing I can think is if someone wanted to write tests for something but didn't want to wait for that action to finish.

E.g. a script that spins up nodes and then runs InSpec to verify all are `pending`...it's a stretch.

Would you like me to remove the transitory ones?

Comment (clintoncwolfe) [reply docs/resources/aws_ec2_instance.md]: I think it's fine as-is.  Sorry, not great collaboration to think aloud on a review :/

**Review by clintoncwolfe — APPROVED** [review]

File: docs/resources/aws_ec2_instance.md (hunk 3 above)
Comment (jerryaldrichiii) [reply docs/resources/aws_ec2_instance.md]: I disagree. We're human and I rather think in the open than in public. Especially in this case. I guarantee anyone else looking at this doc would have thought the same thing.

**Pull request merged**